### ISSUE TITLE
[Cloud] Local changes are not synced to cloud

### ISSUE THREAD

**Issue by Stretsh** [dexie/Dexie.js#1903]
My setup: I am developing my app on my local machine. I have a web server on my LAN, where I push my Nuxt 3 app to.

Problem: Data in my browser for localhost is different from data in my browser for the webserver, which causes the app to fail on the webserver, because the data is not updated.
They both have the same `serverRevision` number, but my local has values in `latestRevisions`, while for the webserver it's just an empty object.

This is what the local syncState looks like:
![dexie-local-syncstate](https://github.com/dexie/Dexie.js/assets/22964062/686cd49a-8c1e-4b4f-b558-3319aa484d29)

And this is the webserver's:
![dexie-webserver-syncstate](https://github.com/dexie/Dexie.js/assets/22964062/2fd79868-ad0a-40b7-b0e7-0582c8a79feb)
(Sorry, Dexie-fu is not strong enough yet to know what parts are sensitive and which not :) )

I added an `await db.cloud.sync()` somewhere in my code, I even tried it with `purpose: 'push'`, but the results are the same.

Am I missing something? Is there a way to trigger/force my local revisions to be pushed to the server? What can I do to prevent these inconstancies in data between two clients?

Of course, please let me know if more details are needed.

**Comment by dfahlander** [comment]
Thanks for the report. Some questions:

1. What version of libraries are you using?
* dexie
* dexie-cloud-addon

2. Is anyone of the servers serving an http address (in-spite of https)? Reason I ask is because some browser features will fail if so, such as WebCrypto and WebSockets. The only exception from not serving over https with a valid certificate corresponding to the DNS name, is to serve from localhost (and also load the page in the browser from http://localhost:<port>).

3. What happens if you delete your database in the browser served from the web server (which seem to be the problematic one) and reload the page and relogin? Does the problem persist every time?

**Comment by Stretsh** [comment]
1: dexie version 4.0.1-beta.6, dexie-cloud-addon 4.0.1-beta.55

2: The webserver is serving over https, localhost is not.

3: On my local machine I have a Chrome tab open for localhost and one for webserver. Since I have some changes in my localhost tab that I need, I only deleted the database on the webserver tab. same result as before after login.
Opened the webserver site on my mobile and get the same data there as in the webserver tab.
I opened localhost in a new Incognito tab and the data there was the same as from the webserver.
This morning I tried both in Edge and both show the data like from the webserver. Even the tab with the localhost version in Edge!

I added a realm in Edge webserver, and that addition is visible in ALL places. Data added from anywhere will be synced to all places, ~~EXCEPT when adding/changing it on my Chrome localhost tab!~~

~~I'm starting to think that this one database, the one I've been using since when I started using Dexie Cloud, is corrupted somehow. It pulls from the cloud, but doesn't seem to be pushing!~~

Wrote all that earlier, then went on a troubleshooting spree... Turns out new updates (add/update/delete) from my Chrome localhost tab ARE synced. There are just two records in the cloud that have been removed from local a long time ago, and those two deletions have not been synced. Not sure what happened back then to prevent the sync from deleting them.

I will keep an eye out for this behavior.
